internal/customgitignore/dir_test.go

```diff
@@ -0,0 +1,732 @@
+package customgitignore_test
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"slices"
+	"strings"
+	"testing"
+
+	"github.com/google/osv-scanner/internal/customgitignore"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing/format/gitignore"
+)
+
+func TestRecursivelyParsingGitignoreFilesFromIgnoredDir(t *testing.T) {
+	t.Parallel()
+
+	// Create a specific git repo with .gitignore files
+	gitRepo := setupGitRepo(t)
+
+	// Read this dir-tree using customgitignore, starting at
+	// the top of the tree at ./dir_a/dir_b
+	start := filepath.Join(gitRepo, "dir_a", "dir_b")
+
+	// Read this dir-tree using customgitignore, starting at the root
+	patterns, _, err := customgitignore.ParseGitIgnores(start, true)
+	if err != nil {
+		t.Errorf("could not read gitignore patterns for test: %v", err)
```

Comment:
I assume we cannot continue with any of these tests if this errors:

```suggestion
		t.Fatalf("could not read gitignore patterns for test: %v", err)
```

(please apply to all tests)